pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -514,7 +514,7 @@ func (c *defaultControl) needsWorkerMigration(o *operation.Operation) (bool, err
 			// If it does still exist then we need to migrate. Otherwise there are no machine resources anymore that
 			// need to be deleted, so no migration would be necessary.
 			machineDeployments := &machinev1alpha1.MachineDeploymentList{}
-			if err := o.K8sSeedClient.Client().List(context.TODO(), &client.ListOptions{Raw: &metav1.ListOptions{Limit: 1}}, &machinev1alpha1.MachineDeploymentList{}); err != nil {
+			if err := o.K8sSeedClient.Client().List(context.TODO(), &machinev1alpha1.MachineDeploymentList{}, client.UseListOptions(&client.ListOptions{Raw: &metav1.ListOptions{Limit: 1}})); err != nil {
```

Comment:
I guess you can use kutils.Limit here is as well 

```suggestion
			if err := o.K8sSeedClient.Client().List(context.TODO(), &machinev1alpha1.MachineDeploymentList{}, client.UseListOptions(kutils.Limit(1)); err != nil {
```

Reply:
okay looks like @ialidzhikov addressed that already.